src/KafkaFlow.Retry.UnitTests/KafkaFlow.Retry/Durable/Polling/QueueTrackerFactoryTests.cs

```diff
@@ -0,0 +1,88 @@
+﻿namespace KafkaFlow.Retry.UnitTests.KafkaFlow.Retry.Durable.Polling
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics.CodeAnalysis;
+    using FluentAssertions;
+    using global::KafkaFlow.Retry.Durable.Definitions;
+    using global::KafkaFlow.Retry.Durable.Encoders;
+    using global::KafkaFlow.Retry.Durable.Polling;
+    using global::KafkaFlow.Retry.Durable.Repository;
+    using global::KafkaFlow.Retry.Durable.Repository.Adapters;
+    using Moq;
+    using Xunit;
+
+    [ExcludeFromCodeCoverage]
+    public class QueueTrackerFactoryTests
+    {
+        public readonly static IEnumerable<object[]> DataTest = new List<object[]>
+        {
+            new object[]
+            {
+                null,
+                Mock.Of<IMessageHeadersAdapter>() ,
+                Mock.Of<IMessageAdapter>() ,
+                Mock.Of<IUtf8Encoder>() ,
+            },
+            new object[]
+            {
+                Mock.Of<IRetryDurableQueueRepository>(),
+                null ,
+                Mock.Of<IMessageAdapter>() ,
+                Mock.Of<IUtf8Encoder>()
+            },
+            new object[]
+            {
+                Mock.Of<IRetryDurableQueueRepository>(),
+                Mock.Of<IMessageHeadersAdapter>() ,
+                null ,
+                Mock.Of<IUtf8Encoder>()
+            },
+            new object[]
+            {
+                Mock.Of<IRetryDurableQueueRepository>(),
+                Mock.Of<IMessageHeadersAdapter>() ,
+                Mock.Of<IMessageAdapter>() ,
+                null
+            }
+        };
+
+        private static readonly RetryDurablePollingDefinition retryDurablePollingDefinition = new RetryDurablePollingDefinition(true, "*/30 * * ? * *", 10, 100, "id");
+
+        [Fact]
+        public void QueueTrackerFactory_Create_Success()
+        {
+            // Arrange
+            var factory = new QueueTrackerFactory(
+                Mock.Of<IRetryDurableQueueRepository>(),
+                Mock.Of<IMessageHeadersAdapter>(),
+                Mock.Of<IMessageAdapter>(),
+                Mock.Of<IUtf8Encoder>());
+
+            // Act
+            var queueTracker = factory.Create(retryDurablePollingDefinition, Mock.Of<IMessageProducer>(), Mock.Of<ILogHandler>());
+
+            // Arrange
+            queueTracker.Should().NotBeNull();
+        }
+
+        [Theory]
+        [MemberData(nameof(DataTest))]
+        public void QueueTrackerFactory_Ctor_WithArgumentNull_ThrowsException(
+            object retryDurableQueueRepository,
```

Comment:
These four arguments should be the four concrete objects instead of using `object`.

Reply:
Done